### DISCUSSION TITLE
Need clarification on license and authorship

### DISCUSSION THREAD

**Discussion by penguin359** [stefanhaustein/TerminalImageViewer#138] [Q&A] [answered]
I am currently working on the Debianization of this package, but I need to make sure I have the licensing correct as well as appropriate authorship. For the licensing, the top-level file seems to be pretty clear that it is dual-licensed for GPL3 or ASL2, however, looking in the source code indicates otherwise. `tiv.cpp` seems to agree, but CImg.h seems to be dual-licensed under either `CeCILL-C` or `CeCILL v2.1` which are approximately LGPL and GPL (version not mentioned). Debian is pretty picky that copyright details are correct in their packages.

Also, for authors, they normally expect a name and email. I've found an email for @aaronliu0130, but I have not found an email for @stefanhaustein to include in that field. I just need to get these clarified before I can submit this package. I'd love to get it into Debian Sid before the DIF happens for Ubuntu 24.04 LTS at the end of this month so tiv will be in the next LTS of Ubuntu, but not sure if there's enough time.

**Comment by stefanhaustein** [comment, marked as answer]
Awesome! My email is stefan.haustein@gmail.com. By end of this month, I assume you are referring to February?

CImg is a third party library that we include for convenience. We'll point that out in the license. Is it sufficient to do so by reference in the license file -- or should we include their full license there, too?

**Reply by aaronliu0130** [reply]
I’ve added a note in the README, which I hope to be sufficient.

**Reply by penguin359** [reply]
It seems I kept adding more answers instead of replying to the existing answer. In any case, here's my work so far if you want to check over it:

https://salsa.debian.org/penguin359/tiv

In particular, reviewing the debian/copyright files and debian/control are most useful.

The Debian ITP bug is here:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1062455

**Reply by penguin359** [reply]
OK, and the RFS bug has been sent. Now, it's just waiting for a Debian Developer out there who is willing to sponsor this package. It sounds like that can take a couple of weeks depending on their work schedule. Hopefully, it's sponsors and uploaded to Debian Sid by February 29th which is the Debian Import Freeze for Ubuntu's next LTS. That's when they stop importing new packages from Debian Sid.

**Reply by penguin359** [reply]
Looks like the package got held up due to a license issue. I guess they found a small portion of code under the BSD 3-clause license in the middle of the file and need that clarified before it can be accepted. In any case, I should still have time to get this package in for Ubuntu 24.04.

**Reply by aaronliu0130** [reply]
Hmm, I can't find the message that says there's a license issue.

I don't entirely understand; isn't the BSD 3-clause fully compatible with Apache and GPL?

**Reply by penguin359** [reply]
I (and my sponsor) received a direct email from Thorsten Alteholz <ftpmaster@ftp-master.debian.org> that my package was REJECTED from FTP Master due to the copyright issue and the tiv package disappeared from this list where it was sitting in limbo:

https://ftp-master.debian.org/new.html

I'll submit a new one shortly with a clarification:

https://salsa.debian.org/penguin359/tiv/-/blob/master/debian/copyright?ref_type=heads

The full text of the rejection follows:

```
Hi,

please also mention the BSD license for parts of
  TerminalImageViewer-1.2.1/src/tiv.cpp
in your debian/copyright.

Thanks!
 Thorsten
```

**Reply by aaronliu0130** [reply]
Hmm, just the mention should be enough indeed. Thanks for your work!

**Reply by penguin359** [reply]
OK, it's on unstable now. It should enter Debian testing as well as Ubuntu in the next couple of weeks.

https://packages.debian.org/sid/tiv

**Reply by penguin359** [reply]
And now it's been pulled into Ubuntu just before the freeze of 24.04 LTS:

https://packages.ubuntu.com/noble/tiv

Next up, Fedora I guess...

**Comment by penguin359** [comment]
I think what will be simplest for Debian Free Software Guidelines will be to just strip out the CImg.h and use the one provided by the existing package:

https://packages.debian.org/bookworm/cimg-dev

I just did a test and it seems to compile without errors using that package. I did have to patch the `Makefile` to remove it's dependency on the internal header file to build it, but it works after compilation. I'll just add that header to the `Files-excluded:` header in the `debian/copyright` file so it strips that file out of upstream when generating the Debian source tarball. This is a pretty common practice in Debian, but they do encourage upstream authors to keep their releases clean when possible. Since CImg is already packaged for Debian and seems to work, this should be simple to get approved after its stripped with that header directive.

**Comment by penguin359** [comment]
OK, the process has started:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1062455

**Comment by stefanhaustein** [comment]
IIRC, there are some error code #defines replicated for windows, copied
from some file under BSD.

Not sure if constants are "copyrightable", but the comments probably are.

I think we should re-type them with different clear names (EXITCODE_XYZ;
keeping the numbers) that don't need a comment and then use them on all
platforms (dropping the corresponding import on linux/macos)

Am Mi., 14. Feb. 2024 um 01:26 Uhr schrieb Loren M. Lang <
***@***.***>:

**Reply by penguin359** [reply]
Yes, I think is on the same level as the Oracle vs. Google lawsuit where they rules that APIs are not copyrightable or something like that. However, it was flagged by one of the gatekeepers and I don't know if I want to get into an in-depth copyright discussion with them. He's just asked that I include the correct information in the copyright file for the current release, 1.2.1, that I am trying to publish.
